rspace-bench/src/test/scala/coop/rchain/rspace/bench/LoopContractBench.scala

```diff
@@ -0,0 +1,110 @@
+package coop.rchain.rspace.bench
+
+import java.util.concurrent.TimeUnit
+
+import coop.rchain.catscontrib.TaskContrib._
+import coop.rchain.rspace.bench.wide.WideBenchState
+import org.openjdk.jmh.annotations.{
+  Benchmark,
+  BenchmarkMode,
+  Fork,
+  Measurement,
+  Mode,
+  OutputTimeUnit,
+  Scope,
+  State,
+  Threads,
+  Warmup
+}
+import org.openjdk.jmh.infra.Blackhole
+
+class LoopContractBench {
+  @Benchmark
+  @BenchmarkMode(Array(Mode.SingleShotTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Threads(1)
+  @Warmup(iterations = 2)
+  @Measurement(iterations = 5)
+  def loopOne(bh: Blackhole, state: LoopOneContractBenchState): Unit = {
+    implicit val scheduler = state.scheduler
+    val result             = state.execute.unsafeRunSync
+    assert(result.isEmpty)
+    bh.consume(processErrors(result))
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.SingleShotTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Threads(1)
+  @Warmup(iterations = 2)
+  @Measurement(iterations = 5)
+  def loopTwo(bh: Blackhole, state: LoopTwoContractBenchState): Unit = {
+    implicit val scheduler = state.scheduler
+    val result             = state.execute.unsafeRunSync
+    assert(result.isEmpty)
+    bh.consume(processErrors(result))
```

Comment:
I think those 4 lines are duplicated in every single `@Benchmark` using RhoBenchBaseState. Could we have them in a single method?

Reply:
let me try, I think I can make it work